pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -190,31 +204,43 @@ func buildHTTPVirtualServices(obj config.Config, gateways map[parentKey]map[k8s.
 
 	httproutes := []*istio.HTTPRoute{}
 	hosts := hostnameToStringList(route.Hostnames)
-	for _, r := range route.Rules {
+	for i := 0; i < len(route.Rules); i++ {
```

Comment:
```suggestion
	for i := range route.Rules {
```

Reply:
This is intentional because `range` can not reach the index appended in the loop
For example:
```go
for i := range arr {
	if i == 0 {
		arr = append(arr, 3)
	}
	println(arr[i])
}
```
It will not print the appended "3", but it will if using `i < len(arr)`